### PR TITLE
Add scm runtime versioning and change to a runtime dependency on GMTK

### PR DESCRIPTION
**Pull request by EricR86**
This pull request addresses some issues with current Segway distributions and adds some features.

- Addition of `setuptools_scm` to enable runtime versioning (no hardcoded values) based on last Git tag and current commit.
```bash
$ segway --version                                                                                                                                                                                        segway 3.0.2.dev0+g41b154e.d20200730 
```
- Removes GMTK as an installation-required dependency, and instead does the check after option parsing.
  - Allows for segway `--version` and `--help` to run without a dependency on GMTK
  - Allows for installation without GMTK required
    - Allows for acquiring the current Segway version without GMTK being installed
    - Allows to be built on readthedocs without GMTK or a hardcoded version
  - Allows for distribution builds without GMTK
- Fixes missing non-python files both through `MANIFEST.in` on a best-effort basis (difficult to maintain longterm) and on `setuptools_scm` when possible through files tracked on Git (which is automatically updated).
  - `setuptools_scm` successfully adds all testcases and files into the source distribution
  - This fixes Bioconda issues where the LICENSE was erroneously omitted amongst some other non-python/data files.

### CODE REVIEW

**Comment by EricR86** [comment]
@michaelmhoffman this is ready for your review

File: doc/conf.py [hunk 1]
```diff
@@ -12,12 +12,11 @@
 # All configuration values have a default; values that are commented out
 # serve to show the default.
 
+from pkg_resources import get_distribution
```
Comment (michaelmhoffman) [inline doc/conf.py:15]: A third-party package, not a stdlib package.

File: docker/Dockerfile [hunk 2]
```diff
@@ -5,27 +5,33 @@ FROM ubuntu:latest
...
     python3-pip \
     libhdf5-dev \
     hdf5-tools \
-    mercurial \
+    git \
     curl \
+&& apt-get clean \
 && rm -rf /var/lib/apt/lists/*
+# Install pip for Python 2
+RUN curl https://bootstrap.pypa.io/get-pip.py --output get-pip.py && python2 get-pip.py
```
Comment (michaelmhoffman) [inline docker/Dockerfile:24]: Can you please add a comment explaining why this is necessary

File: segway/run.py [hunk 3]
```diff
@@ -4201,10 +4209,62 @@ def parse_options(argv):
...
+    # GMTK 1.4.0 (Mercurial id: bdf2718cc6ce tip checkin date: Thu Jun 25 12:31:56 2015 -0700)
+
+    # Try to open gmtkPrint to get the version
+    try:
+        # blocks until finished
+        output_string = subprocess.check_output(["gmtkPrint", "-version"])
+    # If GMTK was not found
+    except OSError:
+        # Raise a runtime error stating that GMTK was not found on the path
+        raise RuntimeError("GMTK cannot be found on your PATH.\nPlease "
```
Comment (michaelmhoffman) [inline segway/run.py:4229]: Can you please have a break in source line where you have the text newline?

File: segway/run.py [hunk 4]
```diff
@@ -4201,10 +4209,62 @@ def parse_options(argv):
...
+
+    # Older versions:
+    # GMTK 1.4.0 (Mercurial id: bdf2718cc6ce tip checkin date: Thu Jun 25 12:31:56 2015 -0700)
+
+    # Try to open gmtkPrint to get the version
+    try:
+        # blocks until finished
+        output_string = subprocess.check_output(["gmtkPrint", "-version"])
+    # If GMTK was not found
+    except OSError:
```
Comment (michaelmhoffman) [inline segway/run.py:4227]: Please check `errno` also for `OSError`

File: segway/run.py [hunk 5]
```diff
@@ -4201,10 +4209,62 @@ def parse_options(argv):
...
+    # Mercurial id: 8995e40101d2 tip
+    # checkin date: Fri Oct 30 10:51:44 2015 -0700
+
+    # Older versions:
+    # GMTK 1.4.0 (Mercurial id: bdf2718cc6ce tip checkin date: Thu Jun 25 12:31:56 2015 -0700)
+
+    # Try to open gmtkPrint to get the version
+    try:
+        # blocks until finished
+        output_string = subprocess.check_output(["gmtkPrint", "-version"])
```
Comment (michaelmhoffman) [inline segway/run.py:4225]: Can you use `optbuild` for this instead? I don't think we use raw `subprocess` anywhere else in Segway and would be good to be consistent.

Comment (michaelmhoffman) [inline segway/run.py:4225]: `output_string` -> `output`

File: segway/run.py [hunk 6]
```diff
@@ -4201,10 +4209,62 @@ def parse_options(argv):
...
+    # If GMTK was not found
+    except OSError:
+        # Raise a runtime error stating that GMTK was not found on the path
+        raise RuntimeError("GMTK cannot be found on your PATH.\nPlease "
+                           "install GMTK from "
+                           "http://melodi.ee.washington.edu/gmtk/ "
+                           "before running Segway.")
+
+    output_lines = output_string.splitlines()
```
Comment (michaelmhoffman) [inline segway/run.py:4234]: `output_lines` -> `lines`

File: segway/run.py [hunk 7]
```diff
@@ -4201,10 +4209,62 @@ def parse_options(argv):
...
+    if len(output_lines) == 1:
+        version_word_index = 1
+    else:
+        version_word_index = -1
+
+    # Get the first line of output
+    first_output_line = output_lines[0].decode()
+
+    # Get the version string from the proper word on the line
+    current_version_string = first_output_line.split()[version_word_index]
```
Comment (michaelmhoffman) [inline segway/run.py:4246]: We usually use `text` instead of `string` for variables, right?

Comment (michaelmhoffman) [reply segway/run.py:4246]: You can abbreviate `current` as `curr`

File: segway/run.py [hunk 8]
```diff
@@ -4201,10 +4209,62 @@ def parse_options(argv):
...
+    output_lines = output_string.splitlines()
+
+    # Check if there's only one line of output (for older versions)
+    if len(output_lines) == 1:
+        version_word_index = 1
+    else:
+        version_word_index = -1
+
+    # Get the first line of output
+    first_output_line = output_lines[0].decode()
```
Comment (michaelmhoffman) [inline segway/run.py:4243]: Why are you doing a decode here? This is weird, seems like any decoding if necessary should be done as soon as the bytes are read in.

File: segway/run.py [hunk 9]
```diff
@@ -4201,10 +4209,62 @@ def parse_options(argv):
...
+
+    # Get the version string from the proper word on the line
+    current_version_string = first_output_line.split()[version_word_index]
+
+    # Get the version number to compare with the minimum version
+    current_version = map(int, current_version_string.split("."))
+    version_zip = zip(current_version, MINIMUM_GMTK_VERSION)
+    for current_version_number, minimum_version_number in version_zip:
```
Comment (michaelmhoffman) [inline segway/run.py:4251]: `for curr_version_item, min_version_item in version_zip`

File: segway/run.py [hunk 10]
```diff
@@ -4201,10 +4209,62 @@ def parse_options(argv):
...
+
+    # Get the first line of output
+    first_output_line = output_lines[0].decode()
+
+    # Get the version string from the proper word on the line
+    current_version_string = first_output_line.split()[version_word_index]
+
+    # Get the version number to compare with the minimum version
+    current_version = map(int, current_version_string.split("."))
+    version_zip = zip(current_version, MINIMUM_GMTK_VERSION)
```
Comment (michaelmhoffman) [inline segway/run.py:4250]: Can you use [`packaging.version`](https://packaging.pypa.io/en/latest/version/) instead? It should be installed with `setuptools`.

File: setup.py [hunk 11]
```diff
@@ -20,18 +17,12 @@
...
-Segway requires GMTK version %s or later to be installed.
-Please update your GMTK version."""
-
 doclines = __doc__.splitlines()
 name, short_description = doclines[0].split(": ")
 long_description = "\n".join(doclines[2:])
 
-url = "http://pmgenomics.ca/hoffmanlab/proj/%s/" % name.lower()
-download_url = "%ssrc/%s-%s.tar.gz" % (url, name, __version__)
+url = "https://pmgenomics.ca/hoffmanlab/proj/%s/" % name.lower()
```
Comment (michaelmhoffman) [inline setup.py:24]: `"https://%s.hoffmanlab.org/" % name`

File: setup.py [hunk 12]
```diff
@@ -20,18 +17,12 @@
...
+download_url = "https://pypi.python.org/pypi/segway"
```
Comment (michaelmhoffman) [inline setup.py:25]: `"https://pypi.python.org/pypi/%s" % name`

**Pull request merged**